eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -169,6 +174,21 @@ jobs:
         vcpkg.deps: 'cmocka'
         VCPKG_DEFAULT_TRIPLET: 'x64-osx'
         build.args: ' -DUNIT_TESTING=ON -DBUILD_PRECONDITIONS=OFF'
+
+      # Build with no logging
+      # WARNINGS_AS_ERRORS          ON
```

Comment:
Nit: Is there a reason to keep comments that may run out of sync with the code below?
E.g. Platform doesn’t seem to be the same as your comment.

Reply:
was just keeping in sync with what we already have. @danieljurek , do we want to keep these as comments? Do other languages/pipelines have this?
I am fine wither way, but what you saying is also true, comments may get out of sync with the actual config, and maybe there isn't much sense to repeat in comments what you can just see several lines below.
I don't see where the Platform mismatches the comment, could you please show me? :)

Reply:
Let's keep them here for now and synchronized. I'm going to be refactoring the pipelines to make better use of agent resources (https://github.com/Azure/azure-sdk-for-c/issues/669) and I want to make sure I'm bucketing these appropriately. 

Other languages do not currently keep these kinds of comments because they're testing only different OS platforms instead of OS platforms AND build configurations.